crates/sui-source-validation-service/src/lib.rs

```diff
@@ -69,6 +63,18 @@ pub fn parse_config(config_path: impl AsRef<Path>) -> anyhow::Result<Config> {
     Ok(toml::from_str(&contents)?)
 }
 
+pub fn repo_name_from_url(url: &str) -> anyhow::Result<String> {
+    let repo_url = Url::parse(url)?;
+    let Some(components) = repo_url.path_segments().map(|c| c.collect::<Vec<_>>()) else {
+	    bail!("Could not discover repository path in url {url}")
+	};
+    let Some(repo_name) = components.last() else {
+	    bail!("Could not discover repository name in url {url}")
+    };
+
```

Comment:
Note that the return value of [`path_segments`](https://docs.rs/url/latest/url/struct.Url.html#method.path_segments) is a [`DoubleEndedIterator`](https://doc.rust-lang.org/nightly/core/str/struct.Split.html#impl-DoubleEndedIterator-for-Split%3C'a,+P%3E) so you don't need to materialize it into a vector to get the last element:

```suggestion
    let Some(components) = repo_url.path_segments() else {
	    bail!("Could not discover repository path in url {url}")
	};
    let Some(repo_name) = components.next_back() else {
	    bail!("Could not discover repository name in url {url}")
    };

```

Reply:
TIL, very nice! This would make a fun rust lint